### PR TITLE
[MOS-549] Linux simulator crashes on GCC12

### PR DESCRIPTION
**Pull request by dawidwojtas-mudita**
Add wrapper for posix stat function
in order to invoke our implementation.

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible.
- [x] Has documentation updated

### CODE REVIEW

**Review by pholat — APPROVED** [review]
rem commented out code :)

File: board/linux/libiosyscalls/src/syscalls_common.hpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+extern "C"
+{
+    int _iosys_stat(const char *file, struct stat *pstat);
+}
```
Comment (Maciej-Mudita) [inline board/linux/libiosyscalls/src/syscalls_common.hpp:9]: please add new line at end of file

File: board/linux/libiosyscalls/src/syscalls_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#include "syscalls_common.hpp"
+
+extern "C"
+{
+    // Replace ASAN stat to our stat
+    int stat(const char *file, struct stat *pstat)
+    {
+        return _iosys_stat(file, pstat);
+    }
+}
```
Comment (Maciej-Mudita) [inline board/linux/libiosyscalls/src/syscalls_wrapper.cpp:13]: same here - no new line

File: board/linux/libiosyscalls/src/syscalls_common.hpp (hunk 1 above)
Comment (dawidwojtas-mudita) [reply board/linux/libiosyscalls/src/syscalls_common.hpp]: Done

**Review by Maciej-Mudita — APPROVED** [review]

**Pull request merged**